status_and_problems.tex

```diff
@@ -0,0 +1,132 @@
+
+The Jenkins scripts also are an important part of the tooling, facilitating a large quantity of actions, like, test a ticket branch, automate releases, or provide periodic build (weekly/daily).
+
+The tools picture is completed by \textit{codekit}, which permits interaction with multiple GitHub packages.
+Just as an example, it is used to create the same tag on all the GitHub packages composing a defined product automatically, instead of creating it manually.
+
+\textit{TO BE CLARIFIED: which are java build tools?}
+
+
+\subsection{Open Problems} \label{sec:openProblems}
+
+In this section a list of problems derived from the current development approach are enumerated. 
+
+While these problems are unresolved the only release process suitable for DM is to release the entire codebase each time.
+
+Their resolution may lead to a different development approach, and therefore the proposed release procedure may need to be reviewed accordingly.
+
+
+\subsubsection{SW Product Composition} \label{sec:problemId}
+
+A GitHub package may be included in more than one software product.
+
+This make impossible to apply in a consistent way the release procedure to a SW products that is not the Science Pipelines.
+The reason is explained in the following example.
+
+Product A is composed by 100 Git packages.
+
+Product B is composed by 80 Git packages.
+
+70 Git packages are shared by both products.
+
+When release 1.0 is done on product A, all 100 packages will be tagged, and corresponding Eups packages are created. This will include the 70 shared packages.
+
+One week lather, product B is ready for the release 1.0, and some of the 70 shared packages have been updated.
+For them, the 1.0 tag required for product B release, will not be the same as the 1.0 tag required for the product A release.
+This implies that for some packages, release 1.0 of product A, is different from release 1.0 of product B.
+
+\textbf{Requirement}: there shall be a 1 to 1 correspondence between software products and GitHub packaged. If this is not possible, each GitHub package shall be included in only one software product.
+Note also that, each GitHub package used for building, unit testing and packaging the software products shall not be included in the SW product itself, but versioned separately and be part of the environment definition. See section \ref{sec:swprod}.
+
+
+\subsubsection{Code Fragmentation} \label{sec:problemCode}
+
+The high number of packages causes problems in that it :
+
+\begin{itemize}
+\item increases the build time: each Git package needs to be build each time.
+\item increases the release time: all Git packages need to be released each time.
+\item increases the failure probability: the tooling may be affected by network glitch or similar technical issues. This may lead to the failure of the build/release process in a non deterministic way.
+\end{itemize}
```

Comment:
I don't think the case has been made that any of these issues, except _maybe_ the third, are significant.

Reply:
In the context of doing a release, all these problems are valid. At the moment.

Reply:
I think you need to justify this. There may be some management advantages to consolidating the code into a smaller number of repositories, but why would it make a substantial difference to build time?

Reply:
Build and unit test time may be proportional to the amount of code in the repository. However, there are other activities done each time a build is done, like for example prepare the repository. I agree this time may not be significant, it is usually few seconds per repository, but in the case of science pipelines, where now we have more than 100 repositories, this starts to be a considerable time. 
In addition, each repository needs to be packaged and pushed to eups.lsst.codes. 

So more are the repositories to consider, and more time will take the complete build process.

For sure, if we have only 10 packages, adding one will not be an issue.
But letting the number of packages grow without any control, this will be an issue.